gui/widget.cpp

```diff
@@ -51,9 +51,7 @@ Widget::Widget(GuiObject *boss, const Common::String &name, const Common::U32Str
 }
 
 void Widget::init() {
-	// Insert into the widget list of the boss
-	_next = _boss->_firstWidget;
-	_boss->_firstWidget = this;
+	_next = _boss->addChild(this);
```

Comment:
maybe the addChild function should call setNext on us, instead of doing it here?

Reply:
Could be done as a future enhancement